### PR TITLE
CASSANDRA-18187 - Add unit tests for per-row TTL and Timestamp usage in CQLSSTableWriter

### PR DESCRIPTION
**Pull request by JeetKunDoug**
CQLSSTableWriter supports per-row setting of both timestamp and TTL values, but it’s not tested or documented today. Add tests to cover setting both TTL and Timestamp values for rows using the CQLSSTableWriter.

### CODE REVIEW

File: src/java/org/apache/cassandra/io/sstable/CQLSSTableWriter.java [hunk 1]
```diff
@@ -247,7 +246,7 @@ public CQLSSTableWriter rawAddRow(List<ByteBuffer> values)
         ClientState state = ClientState.forInternalCalls();
         List<ByteBuffer> keys = modificationStatement.buildPartitionKeyNames(options, state);
 
-        long now = currentTimeMillis();
+        long now = Clock.Global.currentTimeMillis();
```
Comment (dcapwell) [inline src/java/org/apache/cassandra/io/sstable/CQLSSTableWriter.java]: please revert this change, the common style is to static import `Global`, so should remove this style change

File: test/unit/org/apache/cassandra/io/sstable/CQLSSTableWriterTest.java [hunk 2]
```diff
@@ -1116,6 +1118,122 @@ public void testFrozenSetTypeCustomOrdered() throws Exception
...
+    public void testWriteWithTimestamps() throws Exception
+    {
+        long now = currentTimeMillis();
+        long then = now - 1000;
+        final String schema = "CREATE TABLE " + qualifiedTable + " ("
+                              + "  k int,"
+                              + "  c1 int,"
+                              + "  c2 int,"
+                              + "  v text,"
+                              + "  PRIMARY KEY (k)"
```
Comment (dcapwell) [inline test/unit/org/apache/cassandra/io/sstable/CQLSSTableWriterTest.java]: this is confusing, `c1` and `c2` normally means "clustering column" when you read the code, but this does not include it; can you rename these to v1, v2, and v3?  or change the test to make these clustering columns?

File: test/unit/org/apache/cassandra/io/sstable/CQLSSTableWriterTest.java [hunk 3]
```diff
@@ -1116,6 +1118,122 @@ public void testFrozenSetTypeCustomOrdered() throws Exception
...
+    }
+    @Test
+    public void testWriteWithTtl() throws Exception
+    {
+        final String schema = "CREATE TABLE " + qualifiedTable + " ("
+                              + "  k int,"
+                              + "  c1 int,"
+                              + "  c2 int,"
+                              + "  v text,"
+                              + "  PRIMARY KEY (k)"
```
Comment (dcapwell) [inline test/unit/org/apache/cassandra/io/sstable/CQLSSTableWriterTest.java]: this is confusing, `c1` and `c2` normally means "clustering column" when you read the code, but this does not include it; can you rename these to v1, v2, and v3?  or change the test to make these clustering columns?

File: test/unit/org/apache/cassandra/io/sstable/CQLSSTableWriterTest.java [hunk 4]
```diff
@@ -1116,6 +1118,122 @@ public void testFrozenSetTypeCustomOrdered() throws Exception
...
+    }
+    @Test
+    public void testWriteWithTimestampsAndTtl() throws Exception
+    {
+        final String schema = "CREATE TABLE " + qualifiedTable + " ("
+                              + "  k int,"
+                              + "  c1 int,"
+                              + "  c2 int,"
+                              + "  v text,"
+                              + "  PRIMARY KEY (k)"
```
Comment (dcapwell) [inline test/unit/org/apache/cassandra/io/sstable/CQLSSTableWriterTest.java]: this is confusing, `c1` and `c2` normally means "clustering column" when you read the code, but this does not include it; can you rename these to v1, v2, and v3?  or change the test to make these clustering columns?

**Review by dcapwell — CHANGES_REQUESTED** [review]
would also be nice to add tests for statics and collections, that is not a requirement (these tests are fine) but would be nice

File: src/java/org/apache/cassandra/io/sstable/CQLSSTableWriter.java (hunk 1 above)
Comment (JeetKunDoug) [reply src/java/org/apache/cassandra/io/sstable/CQLSSTableWriter.java]: 👍
